awx/plugins/inventory/foreman.py

```diff
@@ -203,6 +203,19 @@ def _get_json(self, url, ignore_errors=None, params=None):
                 break
         return results
 
+    def _post_request(self, url=None, params=None):
+        url = "%s/ansible/api/v2/ansible_inventories/schedule_report" % self.foreman_url
+        session = self._get_session()
+        params = {'input_values': {'required':['all']}}
+        ret = session.post(url, json=params)
+        url = "{0}/api/v2/report_templates/153-Ansible%20Inventory/report_data/{1}".format(self.foreman_url, ret.json().get('job_id'))
+        while True:
+            ret = session.get(url)
+            if ret.status_code != 204:
+                break
+            sleep(5)
```

Comment:
I'm going to have question the reason for this sleep

Reply:
The change we are making with this PR is that we are introducing a choice for tower to fetch inventory by generating an inventory report in foreman. The flow is:
1. Call API to start generating report in foreman.
2. Step 1 returns the job id which we poll at fixed intervals to check when it's complete.
While the job is running in foreman, the API returns status 204. 
3. Once the job completes in foreman, we get a response in our next call to the API and we move ahead to process it.